### PR TITLE
SALTO-6672: Add policy priority E2E

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Add policy priority E2E for policy and policy rules

---

_Additional context for reviewer_
added coverage for password policy and access policy for now, will extend later

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/e2e_test/adapter.test.ts [hunk 1]
```diff
@@ -93,10 +97,7 @@ const createInstance = ({
   parent?: InstanceElement
   name?: string
 }): InstanceElement => {
-  const instValues = {
-    ...mockDefaultValues[typeName],
-    ...valuesOverride,
-  }
+  const instValues = _.merge({}, mockDefaultValues[typeName], valuesOverride)
```
Comment (idoamit8) [inline packages/okta-adapter/e2e_test/adapter.test.ts:100]: nit: 
Why are you using `_.mrege` and not spread operator? 
you can also use `  Object.assign({}, mockDefaultValues[typeName], valuesOverride)`
I think shallow copy is enough here and we don't need deep copy. 
but its up to you

File: packages/okta-adapter/e2e_test/adapter.test.ts [hunk 2]
```diff
@@ -173,23 +174,175 @@ const createChangesForDeploy = async (types: ObjectType[], testSuffix: string):
...
+              },
+            ],
+          },
+        },
+      },
+    },
+    parent: accessPolicy,
+  })
+  const accessPolicyRulePriority = createInstance({
+    typeName: 'AccessPolicyRulePriority',
```
Comment (idoamit8) [inline packages/okta-adapter/e2e_test/adapter.test.ts:249]: Can we use `ACCESS_POLICY_RULE_PRIORITY_TYPE_NAME` const from `constants.ts`?

File: packages/okta-adapter/e2e_test/adapter.test.ts [hunk 3]
```diff
@@ -173,23 +174,175 @@ const createChangesForDeploy = async (types: ObjectType[], testSuffix: string):
...
+  const passwordPolicyRuleB = createInstance({
+    typeName: PASSWORD_RULE_TYPE_NAME,
+    types,
+    valuesOverride: {
+      name: createName('passwordRuleB'),
+    },
+    parent: passwordPolicyA,
+  })
+  const passwordPolicyRulePriority = createInstance({
+    typeName: 'PasswordPolicyRulePriority',
```
Comment (idoamit8) [inline packages/okta-adapter/e2e_test/adapter.test.ts:335]: Please use `PASSWORD_RULE_PRIORITY_TYPE_NAME` const

**Review by idoamit8 — APPROVED** [review]
Great Work! 💯 
Approving to save cycle ✅
Added some small comments

File: packages/okta-adapter/e2e_test/adapter.test.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/e2e_test/adapter.test.ts:100]: I added this cause in some cases I only wanted to pass partial path and I wanted it to actually do the merge
(see line 228 in `defaultRule`)

**Pull request merged**